frontend/src/components/PromptingTemplate.tsx

```diff
@@ -0,0 +1,94 @@
+import { Button, Code, Divider, Heading, SkeletonText, Stack, Text, Textarea, useToast } from "@chakra-ui/react";
+import { ChatIcon } from "@chakra-ui/icons";
+import { useEffect, useState } from "react";
+import { api } from "../api/apiService";
+
+import { CodeHighlight } from "../components";
+
+export type PromptingTemplateProps = {
+  apiPath: (prompt: string) => string;
+  heading: string;
+  description: string;
+  textAreaPlaceholder: string;
+  k: string;
```

Comment:
Ich vermute mal, dass du hier `k` als key verwendest um den State zurückzusetzen, wenn man einen anderen Tab auswählt. Du musst das nicht manuell machen. Jedes Element in React hat ein `key` prop. Wenn du also einfach `key` statt `k` verwendest und das `useEffect` in Zeile 21 entfernst, dann wird React sich darum kümmern, dass die Komponente neu gemounted wird, wenn der key sich ändert.